lstchain/calib/camera/utils.py

```diff
@@ -0,0 +1,70 @@
+import tables
+import numpy as np
+
+from lstchain.io.io import dl1_params_tel_mon_ped_key, dl1_params_tel_mon_cal_key
+
+
+def get_bias_and_std(dl1_file):
+    """
+    Function to extract bias and std of pedestal from interleaved events from dl1 file.
+    Parameters
+    ----------
+    input_filename: str
+        path to dl1 file
+    Returns
+    -------
+    bias, std: np.ndarray, np.ndarray
+        bias and std in p.e.
+    """
+    with tables.open_file(dl1_file) as f:
+        ped = f.root[dl1_params_tel_mon_ped_key]
+        ped_charge_mean = np.array(ped.cols.charge_mean)
+        ped_charge_std = np.array(ped.cols.charge_std)
+        calib = f.root[dl1_params_tel_mon_cal_key]
+        dc_to_pe = np.array(calib.cols.dc_to_pe)
+        ped_charge_mean_pe = ped_charge_mean * dc_to_pe
+        ped_charge_std_pe = ped_charge_std * dc_to_pe
+
+    return ped_charge_mean_pe, ped_charge_std_pe
+
+def get_threshold_from_dl1_file(dl1_path, sigma_clean):
+    """
+    Function to get picture threshold from dl1 from interleaved pedestal events.
+    Return modified picture threshold for tailcut cleaning method.
+    Allow cleaning the most noisy pixels (for example around the star location).
+    Threshold for each pixel is define as:
+        threshold = pedestal_bias + sigma * pedestal_std.
+    Recommended threshold for cleaning:
+        galactic source: picture_thresh=8, boundary_thresh=4, sigma=3
+        extragalactic source: picture_thresh=6, boundary_thresh=3, sigma=2.5
+
+    Parameters
+    ----------
+    input_filename: str
+        path to dl1 file
+    sigma_clean: float
+        cleaning level parameter
+    Returns
+    -------
+    picture_thresh: np.ndarray
+        picture threshold calculated using interleaved pedestal events
+    """
+    high_gain = 0
```

Comment:
Better take the HG index from the ctapipe_lst_io.constants module

Reply:
Done